internal/services/elastic/elastic_monitor_resource.go

```diff
@@ -0,0 +1,444 @@
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"elastic_properties": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"elastic_cloud_user": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"email_address": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elastic_cloud_sso_default_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"elastic_cloud_deployment": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"deployment_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"azure_subscription_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elasticsearch_region": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elasticsearch_service_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"kibana_service_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"kibana_sso_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"sku": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"user_info": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"email_address": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.ElasticEmailAddress,
+						},
+					},
+				},
+			},
+
+			"monitoring_status": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"liftr_resource_category": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"liftr_resource_preference": {
```

Comment:
Please remove the properties that won't be referred in other azurerm resources.

Reply:
Will be updated in the next commit